include/libdnf/comps/group/query.hpp

```diff
@@ -75,8 +73,7 @@ class GroupQuery : public libdnf::sack::Query<Group> {
 
     GroupSackWeakPtr sack;
 
-    class Impl;
-    std::unique_ptr<Impl> p_impl;
+    WeakPtrGuard<GroupQuery, false> data_guard;
```

Comment:
I'm quite sure we don't want weak pointers with a guard on queries (they are created on-demand by the API user, as opposed to owned by base and accessed through a getter via a reference, or in our case, the weak pointer).

Hence I think you can drop this guard, as well as `get_weak_ptr()`, the copy constructor (which creates a new instance of `data_guard` in the copy, thus circumventing the guard's purpose, and for completeness, you're missing an assignment operator that'd go with it :slightly_smiling_face: ) and the destructor.

Reply:
I dropped the query weak pointers in the last commit, so this change wouldn't have any effect at the end. But I can do it before that if you think the commits would make more sense that way.